internal/crossdock/client/onewayctxpropagation/behavior.go

```diff
@@ -0,0 +1,105 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package onewayctxpropagation
+
+import (
+	"context"
+
+	"go.uber.org/yarpc"
+	"go.uber.org/yarpc/encoding/raw"
+	"go.uber.org/yarpc/internal/crossdock/client/dispatcher"
+
+	"github.com/crossdock/crossdock-go"
+	opentracing "github.com/opentracing/opentracing-go"
+)
+
+// Run starts the behavior, testing oneway context propagation
+func Run(t crossdock.T) {
+	assert := crossdock.Assert(t)
+	fatals := crossdock.Fatals(t)
+
+	baggage := map[string]string{
+		"hello": "world",
+		"foo":   "bar",
+	}
+
+	// create handler
+	callBackHandler, serverCalledBack := newCallBackHandler()
+	dispatcher, callBackAddr := dispatcher.CreateOneway(t, callBackHandler)
+	defer dispatcher.Stop()
+
+	client := raw.New(dispatcher.ClientConfig("oneway-server"))
+
+	// make call
+	ack, err := client.CallOneway(
+		newContextWithBaggage(baggage),
+		yarpc.NewReqMeta().
+			Procedure("echo/raw").
+			Headers(yarpc.NewHeaders().
+				With("callBackAddr", callBackAddr)),
+		[]byte{})
+
+	fatals.NoError(err, "call to oneway/raw failed: %v", err)
+	fatals.NotNil(ack, "ack is nil")
+
+	// wait for server to call us back
+	gotBaggage := <-serverCalledBack
+	assert.Equal(baggage, gotBaggage, "server baggage: %s", gotBaggage)
+}
+
+// newCallBackHandler creates a oneway handler that fills a channel
+// with the received body
+func newCallBackHandler() (raw.OnewayHandler, <-chan map[string]string) {
+	serverCalledBack := make(chan map[string]string)
+	handler := func(ctx context.Context, reqMeta yarpc.ReqMeta, body []byte) error {
+		serverCalledBack <- extractBaggage(ctx)
+		return nil
+	}
+	return handler, serverCalledBack
+}
+
+func newContextWithBaggage(baggage map[string]string) context.Context {
+	span := opentracing.GlobalTracer().StartSpan("add baggage")
+	for k, v := range baggage {
+		span.SetBaggageItem(k, v)
+	}
+	return opentracing.ContextWithSpan(context.Background(), span)
+}
+
+func extractBaggage(ctx context.Context) map[string]string {
```

Comment:
Why do you need access to baggage explicitly as `map[string]string`? I.e. why not something like

```go
func getBaggage(ctx Context, key string) {
   if span := opentracing.SpanFromContext(ctx); span != nil {
      return span.GetBaggageItem(key)
   }
}
```
?